core/src/main/java/nl/nn/adapterframework/webcontrol/api/BrowseQueue.java

```diff
@@ -54,10 +56,14 @@
 	@Produces(MediaType.APPLICATION_JSON)
 	public Response getBrowseQueue() throws ApiException {
 		Map<String, Object> returnMap = new HashMap<String, Object>();
-
-		List<String> jmsRealms=JmsRealmFactory.getInstance().getRegisteredRealmNamesAsList();
-		if (jmsRealms.size()==0) jmsRealms.add("no realms defined");
-		returnMap.put("jmsRealms", jmsRealms);
+		JmsBrowser<javax.jms.Message> jmsBrowser = getIbisContext().createBeanAutowireByName(JmsBrowser.class);
+		Set<String> connectionFactories = new LinkedHashSet<String>();
+		// connection factories used in configured jmsSenders etc.
+		connectionFactories.addAll(jmsBrowser.getConnectionFactoryFactory().getConnectionFactoryNames());
+		// configured jms realm
+		connectionFactories.addAll(JmsRealmFactory.getInstance().getConnectionFactoryNames());
+		if (connectionFactories.size()==0) connectionFactories.add("no connection factories found");
+		returnMap.put("connectionFactories", connectionFactories);
```

Comment:
I don't see you use the jmsBrowser anywhere else other then to fetch the `JndiConnectionFactoryFactory`
It might be simpler to autowire the required class directly instead.
`getIbisContext().getBean("connectionFactoryFactory", JndiConnectionFactoryFactory.class)`